crates/optimism/txpool/src/interop.rs

```diff
@@ -0,0 +1,40 @@
+//! Additional support for pooled transactions with [`TransactionInterop`]
+
+/// Helper trait that allows attaching a [`TransactionInterop`].
+pub trait MaybeInteropTransaction {
+    /// Attach a [`TransactionInterop`].
+    fn set_interop(&mut self, conditional: TransactionInterop);
+
+    /// Get attached [`TransactionInterop`] if any.
+    fn interop(&self) -> Option<TransactionInterop>;
+
+    /// Helper that sets the conditional and returns the instance again
+    fn with_interop(mut self, interop: TransactionInterop) -> Self
+    where
+        Self: Sized,
+    {
+        self.set_interop(interop);
+        self
+    }
+}
+
+/// Helper to keep track of cross transaction interop validity
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub struct TransactionInterop {
+    /// Unix timestamp until which tx if considered valid by supervisor.
+    ///
+    /// If None - tx is not validated, it should be automatically revalidated by interop tracker.
+    pub timeout: Option<u64>,
+}
+
+impl TransactionInterop {
+    /// Checks if provided timestamp fits into tx validation window
+    pub fn is_valid(&self, timestamp: u64) -> bool {
+        if let Some(timeout) = self.timeout {
+            if timestamp > timeout {
+                return true;
+            }
+        }
+        false
```

Comment:
this should probably just check `timestamp < self.timeout`

but maybe I misunderstood how the timeout works,

is `timeout` rather a delay, and a tx can only be included after or until the the timeout timestamp is reached?

Reply:
oh, my bad